src/util/streamable.py

```diff
@@ -65,7 +65,13 @@ def dataclass_from_dict(klass, d):
             return None
         return dataclass_from_dict(get_args(klass)[0], d)
     elif is_type_Tuple(klass):
-        return tuple(dataclass_from_dict(get_args(klass)[0], item) for item in d)
+        # Type is tuple, can have multiple different types inside
+        i = 0
+        klass_properties = []
+        for item in d:
```

Comment:
Can we add a test for the streamable changes? This is very sensitive code

Reply:
BTW we need to be careful with this, because we now have streamable in production, so we need to make sure nothing breaks.

Reply:
The previous code assumed that Tuples only contained one type, this code checks each entry's type individually